src/lib/libbrowser.js

```diff
@@ -781,11 +780,7 @@ var LibraryBrowser = {
       var data = msg.data['data'];
       if (data) {
         if (!data.byteLength) data = new Uint8Array(data);
-        if (!info.buffer || info.bufferSize < data.length) {
-          if (info.buffer) _free(info.buffer);
-          info.bufferSize = data.length;
-          info.buffer = _malloc(data.length);
-        }
+        info.buffer = _realloc(info.buffer, data.length);
```

Comment:
Does realloc do the equivalent check that the buffer size is < the desired length? (i.e. does it avoid the actual reallocation in the case where the existing buffer is big enough?